app/services/search_service.rb

```diff
@@ -2,21 +2,34 @@
 
 class SearchService < ::WasteCarriersEngine::BaseService
   def run(page:, term:)
-    return [] if term.blank?
+    return response_hash([]) if term.blank?
 
+    @page = page
     @term = term
-    Kaminari.paginate_array(combined_results).page(page)
+
+    response_hash(search_results)
   end
 
   private
 
-  def combined_results
-    search(WasteCarriersEngine::TransientRegistration) + search(WasteCarriersEngine::Registration)
+  def response_hash(results)
+    {
+      count: results.count,
+      results: Kaminari.paginate_array(results).page(@page)
+    }
+  end
+
+  def search_results
+    @_search_results ||= matching_resources.sort_by(&:reg_identifier)
+  end
+
+  def matching_resources
+    search(WasteCarriersEngine::Registration) + search(WasteCarriersEngine::TransientRegistration)
   end
 
   def search(model)
     model.search_term(@term)
-         .order_by("metaData.lastModified": :desc)
+         .limit(100)
```

Comment:
[NIT] Can I suggest we make this a constant?

Reply:
The `100`? I'm not too fussed to be honest, we only use it once and I don't think it'll need to be changed much (if ever).